sql/distsql: TestSorterAgainstProcessor failed

sql/distsql.TestSorterAgainstProcessor [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/13465578?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/13465578?buildTab=artifacts#/) on master @ [e21c0cac5c4d30e28ed619b06c7fd097be86ed77](https://github.com/cockroachdb/cockroach/commits/e21c0cac5c4d30e28ed619b06c7fd097be86ed77):


```
=== RUN   TestSorterAgainstProcessor
    test_log_scope.go:170: test logs captured to: /artifacts/tmp/_tmp/c06d7cbde0396eecf760bef8c1e14239/logTestSorterAgainstProcessor731300419
    test_log_scope.go:81: use -show-logs to present logs inline
--- seed = 2458433813473110306 spillForced = true nCols = 1 K = 25 ---
CREATE TABLE t(a numeric);
INSERT INTO t VALUES
(-8.526598799069625620E+19),
(-0.04133847703391373812),
(320900.2898053933763),
(5.913242406803089631E+28),
(8.754714418368148198E+30),
(-1.0),
(NULL),
(-1),
(0),
(-8853625780676890569),
(23612459999011573.9),
(-8.419054530761376682E+27),
(NULL),
(-5.734661196313655305E+36),
(91851265150074.7506),
(-362.9436175778306425),
(-1.234E+401),
(38582127182.71172493),
(NULL),
(-1),
(-542533537390380655.0),
(-581844823.4805027722),
(-1),
(NULL);
    columnar_operators_test.go:587: different results on row 4;
        expected:
        [-1.0]
        got:
        [-1]
    panic.go:541: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/c06d7cbde0396eecf760bef8c1e14239/logTestSorterAgainstProcessor731300419
--- FAIL: TestSorterAgainstProcessor (0.54s)
```

Parameters:
 - <code>TAGS=bazel,gss</code>
 - <code>stress=true</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/sql-queries
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestSorterAgainstProcessor.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>


Jira issue: CRDB-35242

--- a/pkg/sql/distsql/columnar_utils_test.go
+++ b/pkg/sql/distsql/columnar_utils_test.go
@@ -35,6 +35,7 @@ import (
 	"github.com/cockroachdb/cockroach/pkg/sql/rowenc"
 	"github.com/cockroachdb/cockroach/pkg/sql/rowexec"
 	"github.com/cockroachdb/cockroach/pkg/sql/sem/eval"
+	"github.com/cockroachdb/cockroach/pkg/sql/sem/tree"
 	"github.com/cockroachdb/cockroach/pkg/sql/types"
 	"github.com/cockroachdb/cockroach/pkg/storage"
 	"github.com/cockroachdb/cockroach/pkg/util/randutil"
@@ -271,6 +272,26 @@ func verifyColOperator(t *testing.T, args verifyColOperatorArgs) error {
 				return false, err
 			}
 			return math.Abs(expFloat-actualFloat) < floatPrecision, nil
+		case types.DecimalFamily:
+			// Decimals might have a different number of trailing zeroes, so
+			// they too require special handling.
+
+			// We first try direct string matching. If that succeeds, then
+			// great!
+			if expected == actual {
+				return true, nil
+			}
+			// Now parse both strings as decimals and use decimal-specific
+			// comparison.
+			expDecimal, err := tree.ParseDDecimal(expected)
+			if err != nil {
+				return false, err
+			}
+			actualDecimal, err := tree.ParseDDecimal(actual)
+			if err != nil {
+				return false, err
+			}
+			return tree.CompareDecimals(&expDecimal.Decimal, &actualDecimal.Decimal) == 0, nil
 		default:
 			return expected == actual, nil
 		}

